plugins/outputs/postgresql/postgresql_test.go

```diff
@@ -0,0 +1,623 @@
+package postgresql
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/outputs/postgresql/utils"
+)
+
+func timeout(t *testing.T, dur time.Duration) {
+	timer := time.AfterFunc(dur, func() {
+		t.Errorf("Test timed out after %s", dur)
+		t.FailNow()
+	})
+	t.Cleanup(func() { timer.Stop() })
+}
+
+type Log struct {
+	level  pgx.LogLevel
+	format string
+	args   []interface{}
+}
+
+func (l Log) String() string {
+	return fmt.Sprintf("%s: "+l.format, append([]interface{}{l.level}, l.args...)...)
+}
+
+// LogAccumulator is a log collector that satisfies telegraf.Logger.
+type LogAccumulator struct {
+	logs []Log
+	cond *sync.Cond
+	tb   testing.TB
+}
+
```

Comment:
Is there a reason you can't use `testutil.Logger`?

Reply:
`testutil.Logger` is very different. `testutil.Logger` is just a log formatter. `LogAccumulator` captures the logs, so the tests can inspect them.